cwms_radar_api/src/main/java/cwms/radar/data/dao/TimeSeriesDaoImpl.java

```diff
@@ -0,0 +1,649 @@
+	}
+
+	public void create(TimeSeries input)
+	{
+		dsl.connection(connection -> {
+			CwmsDbTs tsDao = CwmsDbServiceLookup.buildCwmsDb(CwmsDbTs.class, connection);
+
+			int utcOffsetMinutes = 0;
+			int intervalForward = 0;
+			int intervalBackward = 0;
+			boolean versionedFlag = false;
+			boolean activeFlag = true;
+			tsDao.createTsCodeBigInteger(connection, input.getOfficeId(), input.getName(), utcOffsetMinutes,
+					intervalForward, intervalBackward, versionedFlag, activeFlag);
+		});
+	}
+
+	public void store(TimeSeries input)
+	{
+		dsl.connection(connection -> {
+			store(connection, input.getOfficeId(), input.getName(), input.getUnits(), input.getValues());
+		});
+	}
+
+	public void update(TimeSeries input) throws SQLException
+	{
+		String name = input.getName();
+		if(!timeseriesExists(name)){
+			throw new SQLException("Cannot update a non-existant Timeseries. Create " + name + " first.");
+		}
+		dsl.connection(connection -> {
+			store(connection, input.getOfficeId(), name, input.getUnits(), input.getValues());
+		});
+	}
+
+	public void store(Connection connection, String officeId, String tsId, String units,
+						   List<TimeSeries.Record> values) throws SQLException
+	{
+		CwmsDbTs tsDao =  CwmsDbServiceLookup.buildCwmsDb(CwmsDbTs.class, connection);
+
+		final int count = values == null ? 0 : values.size();
+
+		final long[] timeArray = new long[count];
+		final double[] valueArray = new double[count];
+		final int[] qualityArray = new int[count];
+
+		if(values != null && !values.isEmpty())
+		{
+			Iterator<TimeSeries.Record> iter = values.iterator();
+			for(int i = 0; iter.hasNext(); i++)
+			{
+				TimeSeries.Record value = iter.next();
+				timeArray[i] = value.getDateTime().getTime();
+				valueArray[i] = value.getValue();
+				qualityArray[i] = value.getQualityCode();
+			}
+		}
+
+		final Timestamp versionDate = null;  // Where does this come from?
+		final boolean createAsLrts = false;
+
+		long completedAt = tsDao.store(connection, officeId, tsId, units, timeArray, valueArray, qualityArray, count,
+				STORE_RULE, OVERRIDE_PROTECTION, versionDate, createAsLrts);
+	}
+
+	public void delete(String officeId, String tsId)
+	{
+		dsl.connection(connection -> {
+			CwmsDbTs tsDao = CwmsDbServiceLookup.buildCwmsDb(CwmsDbTs.class, connection);
+
+			tsDao.deleteAll(connection, officeId, tsId);
+		});
+	}
+
+	protected BigDecimal retrieveTsCode(String tsId)
+	{
+
+		return dsl.select(AV_CWMS_TS_ID2.AV_CWMS_TS_ID2.TS_CODE).from(
+				AV_CWMS_TS_ID2.AV_CWMS_TS_ID2).where(AV_CWMS_TS_ID2.AV_CWMS_TS_ID2.CWMS_TS_ID.eq(tsId))
+				.fetchOptional(AV_CWMS_TS_ID2.AV_CWMS_TS_ID2.TS_CODE).orElse(null);
+	}
+
+	public boolean timeseriesExists(String tsId)
+	{
+		return retrieveTsCode(tsId) != null;
+	}
+
+	protected Duration getDuration(String tsId)
+	{
+		Long numMinutes = 0L;
+		try
+		{
+			BigDecimal intervalVal = CWMS_TS_PACKAGE.call_GET_INTERVAL(dsl.configuration(), tsId);
```

Comment:
Are you getting the Duration or the interval? I know the java api uses the single "Duration" class for both a window of time and a distance of time between element but the name of the function getDuration implies you should be calling call_GET_DURATION not call_GET_INTERVAL here. If it's actually the timeseries interval it should be named getInterval even though it's returning a java "Duration" object.

Reply:
You are right, should have been getInterval.  I don't think the method was being used. Easiest to just remove it.